### PR TITLE
feat(project): add dated dir logic and format code

### PR DESCRIPTION
**Pull request by jineshpatel2002**
Added dated directory logic and formated code

### CODE REVIEW

File: cli_app/pipeline.py
Comment (th-rutvi) [inline cli_app/pipeline.py:1]: remove return statement from doc string of process_directory function

File: utils/create_dated_directory.py
Comment (th-rutvi) [inline utils/create_dated_directory.py:1]: Don't create separate file for this, add this code to directory_utils only.

File: utils/create_dated_directory.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+based on the current date. The created directories follow a specific naming convention with the
+current date and an execution number.
+
+"""
+
+import datetime as dt
+import os
+
+
+def producer_dated_dir(source_dir: str, target_dir: str) -> tuple:
```
Comment (kamleshshah15) [inline utils/create_dated_directory.py]: this function seems more specific to our application, it is not necessary all time to create dated directory in both source and target directory , also no need for another file add this function to directory_utils package

File: cli_app/pipeline.py [hunk 2]
```diff
@@ -46,6 +47,27 @@ def __init__(self, path: str, file_type: str, filename_format: str):
         self.total_file_count = 0
         self.source_file_count = 0
         self.target_file_count = 0
+        self.source_dated_dir = self.target_dated_dir = None
+
+    def validate_source_directory(self):
```
Comment (kamleshshah15) [inline cli_app/pipeline.py]: here for creating source and target directory, two separate functions are not required , make one function and add into utility

**Review by kamleshshah15 — CHANGES_REQUESTED** [review]

File: utils/directory_utils.py
Comment (th-rutvi) [inline utils/directory_utils.py:1]: rename create_directory_ifnot with create_directory_if-_not_exists

File: utils/directory_utils.py [hunk 3]
```diff
@@ -31,3 +32,40 @@ def move_files(source_directory: str, target_directory: str, files: list):
...
+
+    today_date = dt.datetime.now().strftime("%d%m%Y")
+
+    try:
+        folders_date = list(filter(lambda x: today_date in x, os.listdir(path)))
+        new_dated_folder = os.path.join(
+            path, today_date + "_" + str(len(folders_date) + 1)
+        )
+
+        os.mkdir(path)
```
Comment (kamleshshah15) [inline utils/directory_utils.py]: change path by new_dated_folder in os.mkdir(), otherwise it will raise logical error

**Review by kamleshshah15 — CHANGES_REQUESTED** [review]

File: cli_app/pipeline.py (hunk 2 above)
Comment (kunj-gandhi889) [reply cli_app/pipeline.py]: yes only single function needed

File: utils/create_dated_directory.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    - target_dir (str): The target directory where corresponding dated subdirectories will be created.
+
+    Returns:
+    tuple: A tuple containing the paths of the newly created source and target directories.
+    """
+
+    today_date = dt.datetime.now().strftime("%d%m%Y")
+
+    try:
+        folders_date = list(filter(lambda x: today_date in x, os.listdir(source_dir)))
```
Comment (kunj-gandhi889) [inline utils/create_dated_directory.py]: i don't think so, this is correct logic for identifying exection number, as firstly it checks in both files and directory

> You should first filter out directory from os.listdir, then only do some operation.

Also, just doing brute_force check using `in` is not good, as date can occur in any format like foldername_01022024 which is not something we are looking for

> Instead use regex or startswith to verify correct format

And, finally it also possible to delete some folder in between some execution like 4 and 6, in such case total files will be 5, and execution number will be decided as 6, but dated folder for execution number 6 already exists, hence it is not good to use `len(folders_date)` to count last execution numbers

> Instead, filters current day files using regex as suggested above, and find highest execution number, this ensures execution number is always incremented.

**Review by kunj-gandhi889 — CHANGES_REQUESTED** [review]

File: cli_app/pipeline.py
Comment (th-rutvi) [inline cli_app/pipeline.py:1]: What's need of line no. 88 & 89?

**Review by th-rutvi — APPROVED** [review]

**Review by kunj-gandhi889 — APPROVED** [review]

**Review by kamleshshah15 — APPROVED** [review]

**Pull request merged**